python/lancedb/query.py

```diff
@@ -38,6 +38,9 @@ class Query(pydantic.BaseModel):
     # sql filter to refine the query with
     filter: Optional[str] = None
 
+    # if True then apply the filter before vector search
+    is_prefilter: bool = False
```

Comment:
Very minor nit: do you want to just name this `prefilter` since that is what is used everywhere else?